app/helpers/tickets.js

```diff
@@ -0,0 +1,80 @@
+import { GetTicketsResponse }  from "../middleware/walletrpc/api_pb";
+
+export const TicketTypes = new Map([
+  [GetTicketsResponse.TicketDetails.TicketStatus.UNKNOWN, "unknown"],
+  [GetTicketsResponse.TicketDetails.TicketStatus.UNMINED, "unmined"],
+  [GetTicketsResponse.TicketDetails.TicketStatus.IMMATURE, "immature"],
+  [GetTicketsResponse.TicketDetails.TicketStatus.LIVE, "live"],
+  [GetTicketsResponse.TicketDetails.TicketStatus.VOTED, "voted"],
+  [GetTicketsResponse.TicketDetails.TicketStatus.MISSED, "missed"],
+  [GetTicketsResponse.TicketDetails.TicketStatus.EXPIRED, "expired"],
+  [GetTicketsResponse.TicketDetails.TicketStatus.REVOKED, "revoked"],
+]);
+
+// decodeVoteScript decodes the output script of a vote transaction into the
+// voted choice. Must be passed an array with the binary script. Returns null
+// if the script is not a vote script. Network is the network for which the
+// vote was cast.
+// Testing whether the provided script is a vote script is really basic right
+// now.
+// Reference for what a voting script looks like (as of this writing):
+// https://github.com/decred/dcrd/blob/3f3174c987091b03bb34f1fdf4614d10ce6fbfc5/blockchain/stake/staketx.go#L458
+export function decodeVoteScript(network, outputScript) {
+  if ((outputScript.length < 4) || (outputScript[0] !== 0x6a)) { // 0x6a == OP_RETURN
+    return null;
+  }
+  if ((outputScript.length > 8)) {
+    console.log("Vote script with extended bits not yet supported");
+    return null;
+  }
+
+  let vote = outputScript.slice(2, 4).reduce((a,v,i) => a|(v<<(i*8), 0));
+  let version = outputScript.length > 4 ? outputScript.slice(4,8).reduce((a,v,i) => a|(v<<(i*8)), 0) : 0;
+
+  // TODO: currently hard coded because dcrwallet doesn't return all
+  // agendas (only the active ones). All agendas are needed for historical
+  // votes.
+  let agendas = {
```

Comment:
I'm not sure whether this is a known issue with dcrwallet's grpc or not. It's not ideal to have all the agendas duplicated here, but as long as they aren't being created too fast, shouldn't be an immediate problem.

Reply:
Yah not much we can do at the present moment, i think we are still thinking about a way to "save" past agendas for use later.  Fine for now